src/main/java/net/masterthought/cucumber/json/Element.java

```diff
@@ -163,11 +163,14 @@ private Status calculateStatus(Configuration configuration) {
     }
 
     private Status calculateStepsStatus() {
-        StatusCounter statusCounter = new StatusCounter();
+        // if all the steps passed, then the steps as a whole passed
+        // otherwise, the status of the first non-passing step will roll up to
+        // the 'steps' status
         for (Step step : steps) {
-            statusCounter.incrementFor(step.getStatus());
+            if (step.getStatus() != Status.PASSED)
+                return step.getStatus();
```

Comment:
so if you have steps with statuses PASSED, UNDEFINED, MISSING then it returns UNDEFINED but for PASSED, MISSING, UNDEFINED it returns MISSING so I believe it is not good idea when the final status depends of the order